velox/type/DecimalUtil.h

```diff
@@ -91,28 +116,126 @@ class DecimalUtil {
 
   template <typename T>
   inline static void fillDecimals(
-      T* decimals,
       const uint64_t* nullsPtr,
+      T* decimals,
       const T* values,
       const int64_t* scales,
-      int32_t numValues,
-      int32_t targetScale) {
-    for (int32_t i = 0; i < numValues; i++) {
-      if (!nullsPtr || !bits::isBitNull(nullsPtr, i)) {
-        int32_t currentScale = scales[i];
-        T value = values[i];
-        if constexpr (std::is_same_v<T, std::int64_t>) { // Short Decimal
+      int32_t targetScale,
+      raw_vector<int32_t>& rows,
+      const velox::common::Filter& filter) {
```

Comment:
Template the `Filter` type and make sure the call to `testValues` are inlined in column reader.